meetings/CI-workinggroup/2019-06-26.md

```diff
@@ -0,0 +1,127 @@
+* Dan Bailey (ILM)
+* Jeff Bradley (Dreamworks)
+* Gordon Bradley (Autodesk)
+
+# Agenda & Notes
+
+## ASWF CI Goals for Year 1  [0:00-0:05]
+
+### Timeframes
+
+* CI Platform decision:   May
+    * Implementations ongoing
+    * Azure Pipelines is the consensus and what we are going forward with
+    * Some weakness in documentation, can be hard to find
+    * Andrew: Microsoft suggests that most of your pipeline building should be done in the context-aware editor inside the Azure Pipelines web UI.
+    * But git commit messages don’t add the DCO / signed-by
+    * Can also look at existing azure_pipelines.yml files on GitHub
+    * We need to share project-specifc AP expertise between projects.
+* Project CII badges:  (Security and static analysis) June
+    * Static analysis via CI
+    * Security, CVE reporting and remedies
+* Dependency management: July
+    * Next phase
+
+## CI Updates for Projects [0:05-0:25]
+
+* OpenVDB
+    * Dan: work done trying to use Aloys Docker images is very promising, most issues resolved. Previously using Ubuntu as base image, moving to CentOS 7 caused a few issues related to GLFW, decided to bump up to more recent version.
+    * Just waiting for official ASWF Docker Images to put in PR, waiting for Andrew to push these out.
+    * Azure Pipelines integration is just a basic "hello world" fornow, then targetting VFX Platform 2019, 2018 and Houdini builds.
+    * Haven’t had much time to look at SonarCloud yet, looking at examples from OpenColorIO
+    * Some outstanding CII-related issues, worked with John Mertic, taking conservative approach to stating compliance for some of the items.
+    * Probably won’t be done by SIGGRAPH but will have some progress done.
+    * Big goals for SIGGRAPH for OpenVDB itself.
+    * Daniel: what has OpenVDB settled on for Houdini downloads, pre-install vs installing at run time? Dan: seems like the simplest approach will be to add Houdini version in a private image that can only be accessed if you sign the EULA. Get one private image with Docker Hub for a personal account, but not for an organization account. Can trim away Houdini components that aren’t needed for build / test, support building against 3 separate Houdini versions. Should we sign up for a paid account? Andrew: Azure Pipelines does have a private Docker registry, not sure if that requires a paid account? Aloys: looks like you need paid account for private image repos (and pay by storage amount). Andrew: we have a Nexus3 system in the ASFW CI account, we could use that to run a private repository (we normally run those public). Daniel: given the EULA requirements for Houdini Apprentice, are we giving up the ability to run Houdini build / tests for private / pre-PR builds? Dan: we need an authentication token anyway, private users can provide their own API key. Aloys: already using his own private key into Docker Hub. Main ASWF pipeline should support builds in context of different organizations for testing.
```

Comment:
Just as an aside - I'm taking as an action item to see what we can do here. Will come back after talking with LF legal.

Reply:
Thanks John